components/raftstore/src/store/fsm/peer.rs

```diff
@@ -6292,15 +6296,18 @@ where
         if let Some(ForceLeaderState::ForceLeader { time, .. }) = self.fsm.peer.force_leader {
             // Clean up the force leader state after a timeout, since the PD recovery
             // process may have been aborted for some reasons.
-            if time.saturating_elapsed()
-                > cmp::max(
-                    self.ctx.cfg.peer_stale_state_check_interval.0,
-                    Duration::from_secs(60),
-                )
-            {
+            if time.saturating_elapsed() > self.ctx.cfg.peer_stale_state_check_interval.0 {
                 self.on_exit_force_leader();
             }
         }
+        if let Some(state) = &mut self.fsm.peer.unsafe_recovery_state {
```

Comment:
There is no `on_check_peer_stale_state_tick` in v2. Could you choose another place?

Reply:
I notice there is one tick `PeerTick::CheckPeerStaleState` but not implemented

Reply:
This is not necessary in raftstore v2, leader actively GC removed peers.

Reply:
This is light-wight, could you check it in `check_unsafe_recovery_state`?

Reply:
Can't cause `check_unsafe_recovery_state` won't be called if no write, then the force leader state may be kept forever and forbid future writes which is a deadlock.

Reply:
I see.